scripts/poppunk_extract_distances.py

```diff
@@ -91,14 +105,32 @@ def isolateNameToLabel(names):
             taxon_name = t.label.replace(' ','_')
             tip_index[r_names.index(taxon_name)] = t
 
+    # Load sparse matrix
+    if args.sparse is not None:
+        sparse_mat = sparse.load_npz(args.sparse)
+    else:
+        X = np.load(args.distances + ".npy")
+
     # open output file
     with open(args.output, 'w') as oFile:
-        oFile.write("\t".join(['Query', 'Reference', 'Core', 'Accessory']))
+        # Write header of output file
+        if args.sparse is not None:
+            oFile.write("\t".join(['Query', 'Reference', 'Core']))
+        else:
+            oFile.write("\t".join(['Query', 'Reference', 'Core', 'Accessory']))
         if args.tree is not None:
             oFile.write("\t" + 'Patristic')
         oFile.write("\n")
-        for i, (r_index, q_index) in enumerate(iterDistRows(r_names, q_names, r_names == q_names)):
-            oFile.write("\t".join([q_names[q_index], r_names[r_index], str(X[i,0]), str(X[i,1])]))
-            if args.tree is not None:
-                oFile.write("\t" + str(pdc(tip_index[r_index], tip_index[q_index])))
-            oFile.write("\n")
+        # Write distances
+        if args.sparse is not None:
+            for (r_index, q_index, dist) in zip(sparse_mat.col, sparse_mat.row, sparse_mat.data):
```

Comment:
Alternatively you could call `todense()`, but perhaps the way of iterating just the non-zero entries as here is more useful for debugging with large matrices?